elodie/filesystem.py

```diff
@@ -98,8 +98,8 @@ def get_all_files(self, path, extensions=None):
             subclasses = get_all_subclasses(Base)
             for cls in subclasses:
                 extensions.update(cls.extensions)
-
         for dirname, dirnames, filenames in os.walk(path):
+            dirnames[:] = [d for d in dirnames if d not in excludeDir]
```

Comment:
Instead of matching directories I think matching any part of the file path makes it more flexible.

Reply:
I will need some hints on how to do this then. Any suggestion on what function/method to use, so I can look it up? Which part on what file you think I should have this in?

Reply:
Perhaps something like below?
```
if excludeDir not in os.path.join(dirname, filename)
    yield os.path.join(dirname, filename)
```

Reply:
What's `dirnames` used for? I'm not seeing it referenced in the rest of the function.

I think logic similar to what you proposed should work but it seems like we should do it on the file names and not the directory names. Instead of this could we do something like the comment below?